### PR TITLE
Create model User and Course

### PR DESCRIPTION
**Pull request by leduytungframgia**

### CODE REVIEW

File: db/migrate/20150507063142_create_courses.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CreateCourses < ActiveRecord::Migration
+  def change
+    create_table :courses do |t|
+      t.string :course_code
+      t.string :name
+      t.text :instrucion
+      t.datetime :start_date
+      t.date_time :end_date
```
Comment (vjnhat) [inline db/migrate/20150507063142_create_courses.rb]: date_time to DateTime

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  
+  validates :password, presence:true, length: {minimum: 6}
+  
+  def User.digest string
+    cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST :
+                                                  BCrypt::Engine.cost
+      BCrypt::Password.create(string, cost: cost)
+  end
+    
+    # Returns a random token.
```
Comment (vjnhat) [inline app/models/user.rb]: remove remark

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  def User.new_token
+      SecureRandom.urlsafe_base64
+  end
+    
+  private
+      def downcase_email
+        self.email = email.downcase
+      end
+      
+      # Creates and assigns the activation token and digest.
```
Comment (vjnhat) [inline app/models/user.rb]: remove remark

**Comment by vjnhat** [comment]
ok

File: .gitignore [hunk 4]
```diff
@@ -7,6 +7,7 @@ capybara-*.html
 /public/system
 /coverage/
 /spec/tmp
+/database.yml
```
Comment (vjnhat) [inline .gitignore]: remove this file with git rm

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+                      format: {with: VALID_EMAIL_REGEX},
+                      uniqueness: {case_sensitive: false}
+  has_secure_password
+  
+  validates :password, presence:true, length: {minimum: 6}
+  
+  def User.digest string
+    cost = ActiveModel::SecurePassword.min_cost ? BCrypt::Engine::MIN_COST :
+                                                  BCrypt::Engine.cost
+      BCrypt::Password.create(string, cost: cost)
```
Comment (dieunb) [inline app/models/user.rb]: indent

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+  end
+    
+  def User.new_token
+      SecureRandom.urlsafe_base64
+  end
+    
+  private
+    def downcase_email
```
Comment (dieunb) [inline app/models/user.rb]: indent

File: config/database.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
+# SQLite version 3.x
```
Comment (dieunb) [inline config/database.yml]: don't push this file to repo

File: spring/8aabb8171a6806d58fc42fdf4600d3ed.pid [hunk 8]
```diff
@@ -0,0 +1 @@
+25955
```
Comment (dieunb) [inline spring/8aabb8171a6806d58fc42fdf4600d3ed.pid]: ???

File: db/migrate/20150507063142_create_courses.rb [hunk 9]
```diff
@@ -0,0 +1,13 @@
+class CreateCourses < ActiveRecord::Migration
+  def change
+    create_table :courses do |t|
+      t.string :course_code
+      t.string :name
+      t.text :instrucion
+      t.datetime :start_date
+      t.datetime :end_date
+
```
Comment (vjnhat) [inline db/migrate/20150507063142_create_courses.rb]: creator_id ????

File: app/models/user.rb [hunk 10]
```diff
@@ -0,0 +1,12 @@
+class User < ActiveRecord::Base
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+
+  validates :name, presence:true, length: {maximum: 30}
+  validates :email, presence:true, length: {maximum: 50},
+                      format: {with: VALID_EMAIL_REGEX},
```
Comment (vjnhat) [inline app/models/user.rb]: check remark, it should be: 
 validates :email, presence:true, length: {maximum: 50},
                           format: {with: VALID_EMAIL_REGEX},

File: db/schema.rb [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    t.datetime "end_date"
+    t.datetime "created_at",  null: false
+    t.datetime "updated_at",  null: false
+  end
+
+  create_table "users", force: :cascade do |t|
+    t.string   "name"
+    t.string   "email"
+    t.string   "password"
+    t.boolean  "supervisor"
```
Comment (vjnhat) [inline db/schema.rb]: suppervior?? change to supper please

**Pull request merged**